cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java

```diff
@@ -451,13 +543,74 @@ protected void delete(String namespace, String name, Handler<AsyncResult<Void>>
         }
     };
 
+    private final CompositeOperation<TopicController> updateTopicController = new CompositeOperation<TopicController>() {
+
+        @Override
+        public Future<?> composite(String namespace, ClusterOperation<TopicController> operation) {
+            TopicController topicController = operation.cluster();
+            ClusterDiffResult diff = operation.diff();
+
+            Future<Void> fut;
+            if (diff != null) {
+                fut = patchDeployment(topicController, namespace, diff);
+            } else {
+                // that's the case we are moving from no topic controller deployed to an updated cluster ConfigMap
+                // which contains topic controller configuration for deploying it (so there is no Deployment to patch
+                // but to create)
+                fut = deploymentOperations.create(topicController.generateDeployment());
+            }
+
+            return fut;
+        }
+
+        @Override
+        public ClusterOperation<TopicController> getCluster(String namespace, String name) {
+            ClusterDiffResult diff = null;
+            TopicController topicController = null;
+            ConfigMap tcConfigMap = configMapOperations.get(namespace, name);
+
+            if (tcConfigMap != null) {
+                topicController = TopicController.fromConfigMap(tcConfigMap);
+                if (topicController != null) {
+                    log.info("Updating Topic Controller {} in namespace {}", topicController.getName(), namespace);
+                    Deployment dep = deploymentOperations.get(namespace, topicController.getName());
+                    diff = topicController.diff(dep);
+                }
+            } else {
+                throw new IllegalStateException("ConfigMap " + name + " doesn't exist anymore in namespace " + namespace);
```

Comment:
Presumably a race? Did you observe this in practice?

Reply:
I noticed that the same is in all the other `getCluster` of delete operation maybe someone before me observed this in practice ?